mysql-test/suite/galera_3nodes/t/galera_query_cache.test

```diff
@@ -0,0 +1,132 @@
+--source include/big_test.inc
+--source include/force_restart.inc
+--source include/galera_cluster.inc
+
+#
+# MDEV-28641 : Query cache entries not invalidated on slave of a Galera cluster
+#
+# We use two 2-node galera clusters as follows
+#
+#       A(1) <-> B(2)              {Galera cluster 1}
+#       |                          {Async replication}
+#       D(3) <-> E(4)              {Galera cluster 2}
+#
+# Normal asyncronous replication is used between nodes 1 and 3
+# so that node_1 is master and node_3 a slave.
+#
+# In this test we can't test is some query fast or slow but we can
+# test does all nodes see all rows (this is not true before fix)
+#
+--connect node_3, 127.0.0.1, root, , test, $NODE_MYPORT_3
+--connect node_4, 127.0.0.1, root, , test, $NODE_MYPORT_4
+
+--connection node_2
+call mtr.add_suppression("WSREP: Ignoring server id for non bootstrap node.");
+--connection node_4
+call mtr.add_suppression("WSREP: Ignoring server id for non bootstrap node.");
+
+--connection node_3
+
+--replace_result $NODE_MYPORT_1 NODE_MYPORT_1
+--eval CHANGE MASTER TO master_host='127.0.0.1', master_user='root', master_port=$NODE_MYPORT_1, master_use_gtid=current_pos;
+START SLAVE;
+--source include/wait_for_slave_to_start.inc
+
+--connection node_1
+
+CREATE TABLE t1 (id bigint primary key auto_increment, msg varchar(100)) engine=innodb;
+INSERT INTO t1 SELECT NULL, md5(rand());
+INSERT INTO t1 SELECT NULL, md5(rand()) from t1 LIMIT 1000000;
+INSERT INTO t1 SELECT NULL, md5(rand()) from t1 LIMIT 1000000;
+INSERT INTO t1 SELECT NULL, md5(rand()) from t1 LIMIT 1000000;
+INSERT INTO t1 SELECT NULL, md5(rand()) from t1 LIMIT 1000000;
+INSERT INTO t1 SELECT NULL, md5(rand()) from t1 LIMIT 1000000;
+INSERT INTO t1 SELECT NULL, md5(rand()) from t1 LIMIT 1000000;
+INSERT INTO t1 SELECT NULL, md5(rand()) from t1 LIMIT 1000000;
+INSERT INTO t1 SELECT NULL, md5(rand()) from t1 LIMIT 1000000;
+INSERT INTO t1 SELECT NULL, md5(rand()) from t1 LIMIT 1000000;
+INSERT INTO t1 SELECT NULL, md5(rand()) from t1 LIMIT 1000000;
+INSERT INTO t1 SELECT NULL, md5(rand()) from t1 LIMIT 1000000;
+INSERT INTO t1 SELECT NULL, md5(rand()) from t1 LIMIT 1000000;
+INSERT INTO t1 SELECT NULL, md5(rand()) from t1 LIMIT 1000000;
+INSERT INTO t1 SELECT NULL, md5(rand()) from t1 LIMIT 1000000;
+INSERT INTO t1 SELECT NULL, md5(rand()) from t1 LIMIT 1000000;
+INSERT INTO t1 SELECT NULL, md5(rand()) from t1 LIMIT 1000000;
+INSERT INTO t1 SELECT NULL, md5(rand()) from t1 LIMIT 1000000;
+INSERT INTO t1 values (NULL, 'foobar');
+
+--sync_slave_with_master node_3
+
+#
+# All nodes should see one row and first query is slow and second fast
+#
+--connection node_1
+--echo # node_1
+SELECT * FROM t1 WHERE msg='foobar';
```

Comment:
Selects here and below show also the primary key column, which depends on auto increment settings. This could cause test failure if wsrep auto inc settings change for some reason. Maybe change it to `SELECT msg FROM t1 WHERE msg='foobar'`?